### PR TITLE
Reduce cxxtools::Json footprint 

### PR DESCRIPTION
**Pull request by sinanyavas-eaton**

### CODE REVIEW

**Review by ericclappier-eaton — APPROVED** [review]

File: ecpp/rest_document_GET.ecpp [hunk 1]
```diff
@@ -89,8 +88,9 @@ UserInfo user;
 
     // Send reply.
     reply.setContentType("application/json;charset=UTF-8");
-    cxxtools::JsonSerializer serializer(reply.out());
-    serializer.serialize(si);
+    std::ostringstream output;
+    JSON::writeToStream(output, si, false);
+    reply.out() << output.str();
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/rest_document_GET.ecpp]: simpler approach:
reply.out() << JSON::writeToString(si, false);

File: ecpp/rest_documents_GET.ecpp [hunk 2]
```diff
@@ -121,8 +121,9 @@ UserInfo user;
 
     // Send reply.
     reply.setContentType("application/json;charset=UTF-8");
-    cxxtools::JsonSerializer serializer(reply.out());
-    serializer.serialize(rootSi);
+    std::ostringstream output;
+    JSON::writeToStream(output, rootSi, false);
+    reply.out() << output.str();
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/rest_documents_GET.ecpp]: simpler approach:
reply.out() << JSON::writeToString(rootSii, false);

File: ecpp/rest_documents_POST.ecpp [hunk 3]
```diff
@@ -86,8 +86,9 @@ UserInfo user;
     cxxtools::SerializationInfo si;
     si.addMember("id") <<= newId;
 
-    cxxtools::JsonSerializer serializer(reply.out());
-    serializer.serialize(si);
+    std::ostringstream output;
+    JSON::writeToStream(output, si, false);
+    reply.out() << output.str();
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/rest_documents_POST.ecpp]: simpler approach:
reply.out() << JSON::writeToString(si, false);

**Review by FrancoisRegisDegott-eaton — APPROVED** [review]

File: ecpp/rest_document_GET.ecpp [hunk 4]
```diff
@@ -35,8 +35,7 @@
 #include <fty_security_wallet.h>
 #include <fty_common_rest.h>
 #include <fty_common_socket.h>
-
-#include <cxxtools/jsonserializer.h>
+#include <fty_common_json.h>
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/rest_document_GET.ecpp:38]: #include <cxxtools/serializationinfo.h>  missed

File: ecpp/rest_documents_POST.ecpp [hunk 5]
```diff
@@ -35,7 +35,7 @@
 #include <fty_security_wallet.h>
 #include <fty_common_rest.h>
 #include <fty_common_socket.h>
-#include <cxxtools/jsonserializer.h>
+#include <fty_common_json.h>
```
Comment (FrancoisRegisDegott-eaton) [inline ecpp/rest_documents_POST.ecpp:38]: include cxxtools/serializationinfo.h   ?

**Review by FrancoisRegisDegott-eaton — DISMISSED** [review]

**Pull request merged**